nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/DeleteHBaseRow.java

```diff
@@ -0,0 +1,178 @@
+                }
+
+                for (int index = 0; index < flowFiles.size(); index++) {
+                    session.transfer(flowFiles.get(index), REL_SUCCESS);
+                }
+            } catch (Exception ex) {
+                getLogger().error("");
+                for (int index = 0; index < flowFiles.size(); index++) {
+                    session.transfer(flowFiles.get(index), REL_FAILURE);
+                }
+            }
+        }
+    }
+
+    private void doDeleteFromAttributes(List<FlowFile> flowFiles, ProcessContext context, ProcessSession session, int batchSize) throws Exception {
+        Map<String, List<byte[]>> deletes = new HashMap<>();
+        for (FlowFile flowFile : flowFiles) {
+            String rowKey = context.getProperty(ROW_ID).evaluateAttributeExpressions(flowFile).getValue();
+            String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+            if (rowKey == null || rowKey.trim().equals("")) {
+                throw new RuntimeException("Could not find a row key with provided EL statement, stopping...");
+            }
+
+            List<byte[]> batch = deletes.get(tableName);
+            if (batch == null) {
+                batch = new ArrayList<>();
+                deletes.put(tableName, batch);
+            }
+
+            batch.add(rowKey.getBytes("UTF-8"));
+        }
+
+        processDeletes(deletes, batchSize);
+    }
+
+    private void doDeleteFromBody(List<FlowFile> flowFiles, ProcessContext context, ProcessSession session, int batchSize) throws Exception {
+        Map<String, List<byte[]>> deletes = new HashMap<>();
+        for (FlowFile flowFile : flowFiles) {
+            session.read(flowFile, in -> {
+                String data = IOUtils.toString(in, "UTF-8");
+                String[] lines = data.split("[\\r\\n]");
+                final String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+                for (String line : lines) {
+                    String[] separated = line.split(",");
+                    for (String part : separated) {
+                        List<byte[]> batch = deletes.get(tableName);
+                        if (batch == null) {
+                            batch = new ArrayList<>();
+                            deletes.put(tableName, batch);
+                        }
+                        batch.add(part.trim().getBytes("UTF-8"));
+                    }
+                }
+            });
+        }
+
+        processDeletes(deletes, batchSize);
+    }
+
+    private void processDeletes(Map<String, List<byte[]>> deletes, int batchSize) throws IOException {
+        for (Map.Entry<String, List<byte[]>> entry : deletes.entrySet()) {
+            if (entry.getValue().size() <= batchSize) {
+                clientService.delete(entry.getKey(), entry.getValue());
+            } else {
```

Comment:
I think this is quite a dangerous behavior. This might lead to situation in which part of the flowfiles processed are succeeded and some not. Can't we just submit one batch per run? The user can control the batch size using FLOWFILE_FETCH_COUNT...

Reply:
This follows the pattern of other HBase processors in allowing the table to be configured with EL. So we can't do one big batch since it's possible the user will a group of flowfiles that will go against multiple tables.

Reply:
does this mean that we are telling the user that some flowfile failed even though they were successful? I think the best approach here would be to process one flowfile per time: in this way the user can be sure of what happened. What do you think?

Reply:
I'll have to think about the body scenario, but I don't see a good way to do otherwise if the user batches on multiple flowfiles short of hitting HBase with a ton of one off delete statements. We can do that, but my understanding is that HBase really doesn't like write operations to be handled like that.

Reply:
**batches on multiple flowfiles using attributes instead of the body.

Reply:
yes, I think there isn't a perfect solution. Though, if the user sees that there are performance problems using a lot of small flowfiles, he can always put the keys in the body and merge the flowfiles, thus obtaining better performances. I think that consistency and correctness of the outcome are more important than performance. What do you think?